File: Day_01/Part_01.java
package Day_01;

import java.io.*;
import java.util.*;

public class Part_01 {
    public static void main(String[] args) {
        try {
        
            File inputFile = new File(args[0]);
            Scanner s = new Scanner(inputFile);

            ArrayList<Integer> elves = new ArrayList<Integer>();
            int currentElf = 0;

            // Map each elf from the highest calorie count to the lowest and include their index
            for (int i = 0; s.hasNextLine(); ){
                String line = s.nextLine();
                if (line.length() == 0) {
                    elves.add(currentElf);
                    currentElf = 0;
                    i++;
                    continue;
                } else {
                    currentElf += Integer.parseInt(line);
                }
            }

            int mostCalories = Collections.max(elves);
            int mostCaloriesIndex = elves.indexOf(mostCalories);

            System.out.println(mostCalories);
            System.out.println(mostCaloriesIndex);

            } catch (FileNotFoundException e) {
                System.out.println("File not found");
            }

        
    } // End Main

}

File: Day_01/Part_01.js
const fileUtil = require('../fileUtil')  
const data = fileUtil.syncReadFile('./input.txt');

let elfs = [];
let calorieSum = 0;

data.forEach((line) => {
    if (line !== '') {
        calorieSum += parseInt(line);
    } else {
        elfs.push(calorieSum);
        calorieSum = 0;
    }
});

console.log('Max number of calories: ' + Math.max(...elfs));
console.log('Index of max number of calories: ' + elfs.indexOf(Math.max(...elfs)));